tools/broccoli-ts2dart.ts

```diff
@@ -0,0 +1,52 @@
+/// <reference path="typings/node/node.d.ts" />
+/// <reference path="typings/broccoli.d.ts" />
+
+import fs = require('fs');
+import path = require('path');
+import * as ts2dart from 'ts2dart';
+
+const mkdirp = require('mkdirp');
+
+
+/**
+ * Interface declaration, because broccoli-caching-writer doesn't have a .d.ts file.
+ */
+interface CachingWriter {
+  outputPath: string;
+  inputPaths: string[];
+
+  new (inputNodes: string[], options: any);
+}
+type CachingWriterConstructor = { new (...args: any[]): CachingWriter };
+const CACHING_WRITER_: CachingWriterConstructor = require('broccoli-caching-writer');
+
+
+/**
+ * Class that builds a TS file into its Dart equivalent.
+ * TODO(hansl): remove this and replace it with broccoli-ts2dart from angular/angular.
+ * We need to move broccoli-ts2dart outside of angular/angular, and there's a lot of external
+ * dependencies so creating this was simpler.
+ */
```

Comment:
After talking to @IgorMinar this morning we decided to copy over the files from Angular instead of using this approach. So this PR will change dramatically (I'm thinking of dropping it entirely and making a new PR).